src/dmd/mtype.d

```diff
@@ -4922,6 +4922,28 @@ extern (C++) final class TypeFunction : TypeNext
         return MATCH.nomatch;
     }
 
+    /**
+     * Check if the arguments of a call are possibly inverted and prints a
+     * supplemental error message if the call would compile.
+     *
+     * Params:
+     *      tthis   = The type of the parent aggregate if it's a member function.
+     *      args    = The arguments of the call.
+     *      sc      = The scope of the call.
+     */
+    extern (D) void checkInvertedMatch(Type tthis, Expressions* args, Scope* sc)
+    {
+        if (args is null || args.length != 2)
+            return;
+        Expressions* invertedArgs = new Expressions(2);
+        (*invertedArgs)[0] = (*args)[1];
+        (*invertedArgs)[1] = (*args)[0];
+        if (const m = callMatch(tthis, invertedArgs, 0, null, sc))
+        {
+            errorSupplemental((*args)[1].loc, "Could the order of arguments be inverted ?");
```

Comment:
This could be confusing to some users: how about printing the actual suggested function and use the more common "did you mean ... "?

Reply:
Yes indeed. Originally I used "would". Could gives more the idea of an action... I'd like an non ambiguous translation of _L'ordre des paramètres serait-il inversé ?_.

Reply:
The nuance is rather

- would: maybe it is the case
- could: technically doable ?

Reply:
> would: maybe it is the case

That's not how I interpret it.  When I see "would" I see, "If I did x, would y happen?".  To convey "maybe it is the case" I would use "Might the order of arguments be inverted?" or "Are the order of arguments inverted?"  But, "Could" also conveys this in the dialect of English I learned, but English speakers don't always speak English correctly.

You could also propose with a statement: "The argument order may be inverted".  "Did you mean..." is commonly used in DMD, so maybe "Did you mean to pass the arguments in a different order?"

Reply:
"Did you mean" is just used by the spell checker. I would not use it here because the conviction is way much lower.

rather "Might the order of arguments be inverted?" then.

Reply:
I don't want to start bikeshedding here, but how is this different to the spell checker? It's a suggestion for the user. In either way my main point was that we should print the actual function call in the message.

Reply:
I'm not in favor of formatting the call. My arguments are:

1. For now the information to do so is not there since the error is generated from a TypeFunction, not a CallExpr
2. Previous errors show enough concrete information (formatting of the two arguments in this case) to understand the suggestion.

Please decide and eventually block the PR with the "required changes" system otherwise I won't touch the code.